### PR TITLE
mon: let CrushWrapper::get_validated_type_id() return an optional<>

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-07-21_08:19:42-rados-wip-kefu-testing-2021-07-21-1257-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-07-21_13:52:36-rados-wip-kefu-testing-2021-07-21-2006-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-07-21_14:42:35-rados-wip-kefu-testing-2021-07-21-2006-distro-basic-smithi/

e2e failures tracked by https://tracker.ceph.com/issues/51728

**Comment by tchaikov** [comment]
@neha-ojha @jdurgin @gregsfortytwo ping?

File: src/crush/CrushWrapper.h [hunk 1]
```diff
@@ -374,9 +374,12 @@ class CrushWrapper {
   }
   int get_type_id(const std::string& name) const {
     build_rmaps();
-    if (type_rmap.count(name))
-      return type_rmap[name];
-    return -1;
+    auto found = type_rmap.find(name);
+    if (found != type_rmap.end()) {
+      return found->second;
```
Comment (rzarzynski) [inline src/crush/CrushWrapper.h:379]: Should we tag this with the `performance` label?

**Review by rzarzynski — APPROVED** [review]

File: src/crush/CrushWrapper.h (hunk 1 above)
Comment (tchaikov) [reply src/crush/CrushWrapper.h:379]: sure.

**Pull request merged**